larrecodnn/ImagePatternAlgs/Tensorflow/Modules/EvaluateROIEff_module.cc

```diff
@@ -47,104 +56,433 @@ class nnet::EvaluateROIEff : public art::EDAnalyzer {
+    h_energy[i] = tfs->make<TH1D>(Form("h_energy_%d",i), Form("Plane %d; Energy (MeV);",i), 100,0,1);
+    h_energy_roi[i] = tfs->make<TH1D>(Form("h_energy_roi_%d",i), Form("Plane %d; Energy (MeV);",i), 100,0,1);
+
+    h_purity[i] = tfs->make<TH1D>(Form("h_purity_%d",i), Form("Plane %d; Purity;",i), 20,0,1);
+    h_roi[i] = tfs->make<TH1D>(Form("h_roi_%d",i), Form("Plane %d; Purity;",i), 5,0,5); // index 0: total rois; index 1: total sig rois
+    
+    h1_roi_max[i] = tfs->make<TH1D>(Form("h1_roi_max_%d",i), Form("Plane %d; Max adc;", i), 50,0,50);
+    h1_roi_max_sim[i] = tfs->make<TH1D>(Form("h1_roi_max_sim_%d",i), Form("Plane %d; Max adc;", i), 50,0,50);
+
+    h1_tickdiff_max[i] = tfs->make<TH1D>(Form("h1_tickdiff_max_%d",i), Form("Plane %d; tick diff (maxE - max pulse);", i), 100,-50,50);
   }
-}
 
-void
-nnet::EvaluateROIEff::endJob()
-{
+  h_purity_all = tfs->make<TH1D>("h_purity_all", "All Planes; Purity;", 20,0,1);
+}
 
+void nnet::EvaluateROIEff::endJob(){
   art::ServiceHandle<art::TFileService const> tfs;
 
-  for (int i = 0; i < 3; ++i) {
-    std::cout << i << " " << h_energy[i]->GetNbinsX() << " " << h_energy_roi[i]->GetNbinsX()
-              << std::endl;
-    if (TEfficiency::CheckConsistency(*(h_energy[i]), *(h_energy_roi[i]))) {
+  for (int i = 0; i<3; ++i){
+    std::cout<<i<<" "<<h_energy[i]->GetNbinsX()<<" "<<h_energy_roi[i]->GetNbinsX()<<std::endl;
+    if(TEfficiency::CheckConsistency(*(h_energy_roi[i]), *(h_energy[i]))){
       eff_energy[i] = tfs->make<TEfficiency>(*(h_energy_roi[i]), *(h_energy[i]));
-      eff_energy[i]->Write(Form("eff_energy_%d", i));
+      eff_energy[i]->Write(Form("eff_energy_%d",i));
+    }
+  }
+
+  // purity
+  for (int i=0; i<3; ++i) {
+    if (fCount_Roi_total[i]==0) continue;
+    cout << "\nplane : " << i << endl;
+    cout << "ROI purity (a) sig roi/total rois: " << fCount_Roi_sig[i] << "/" << fCount_Roi_total[i] << " = " << 1.0*fCount_Roi_sig[i]/fCount_Roi_total[i] << endl;
+
+    cout << "ROI purity (b) from histogram: sig/total: " << h_roi[i]->GetBinContent(2) << "/" << h_roi[i]->GetBinContent(1) << " = " << 1.0*(h_roi[i]->GetBinContent(2))/(h_roi[i]->GetBinContent(1)) << endl;
+  }
+
+  cout << "\nROI combined purity (all planes) (a) sig roi/total rois: " << fCount_Roi_sig[0]+ fCount_Roi_sig[1] + fCount_Roi_sig[2]<< "/" << fCount_Roi_total[0]+fCount_Roi_total[1] + fCount_Roi_total[2] << " = " << 1.0*(fCount_Roi_sig[0]+fCount_Roi_sig[1]+fCount_Roi_sig[2])/(fCount_Roi_total[0]+fCount_Roi_total[1]+fCount_Roi_total[2]) << endl;
+
+  cout << "ROI combined purity (all planes) (b) from histogram: " << h_roi[0]->GetBinContent(2) + h_roi[1]->GetBinContent(2) + h_roi[2]->GetBinContent(2) << "/" << h_roi[0]->GetBinContent(1) + h_roi[1]->GetBinContent(1) + h_roi[2]->GetBinContent(1) << " = "<< 1.0*(h_roi[0]->GetBinContent(2) + h_roi[1]->GetBinContent(2) + h_roi[2]->GetBinContent(2)) / (h_roi[0]->GetBinContent(1) + h_roi[1]->GetBinContent(1) + h_roi[2]->GetBinContent(1)) << endl;
```

Comment:
As these lines of code are only for printing out information, I suggest removing them.